### PR TITLE
Allow rejecting invitations

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: specification/client_server_api.rst [hunk 1]
```diff
@@ -931,7 +936,10 @@ Leaving rooms
 
 
 A user can leave a room to stop receiving events for that room. A user must
-have joined the room before they are eligible to leave the room. If the room is
+have been invited to, or joined the room before they are eligible to leave the
```
Comment (Kegsay) [inline specification/client_server_api.rst]: > have been invited to, or joined

have been invited to or have joined

File: specification/client_server_api.rst [hunk 2]
```diff
@@ -931,7 +936,10 @@ Leaving rooms
...
+room. Leaving from an invited state serves to reject the invite.
```
Comment (Kegsay) [inline specification/client_server_api.rst]: > Leaving from an invited state serves to reject the invite.

Leaving a room that the user has been invited to rejects the invite.

File: specification/client_server_api.rst [hunk 3]
```diff
@@ -931,7 +936,10 @@ Leaving rooms
...
+
+Whether or not they actually joined the room, if the room is
 an "invite-only" room, they will need to be re-invited before they can re-join
```
Comment (Kegsay) [inline specification/client_server_api.rst]: Needless comma

**Comment by Kegsay** [comment]
So there are no new endpoints then? I just hit `/leave` on an invited room and it gets rejected?

File: specification/client_server_api.rst (hunk 1 above)
Comment (illicitonion) [inline specification/client_server_api.rst]: Done

File: specification/client_server_api.rst (hunk 2 above)
Comment (illicitonion) [inline specification/client_server_api.rst]: Done

File: specification/client_server_api.rst (hunk 3 above)
Comment (illicitonion) [inline specification/client_server_api.rst]: Done

**Comment by illicitonion** [comment]
Yep, no new endpoints, just overloading /leave

**Comment by illicitonion** [comment]
Merged manually

**Pull request closed without merging**